script/eslint-plugin-va/rules/prefer-web-component-library.js

```diff
@@ -48,6 +48,63 @@ const telephoneTransformer = (context, node) => {
   });
 };
 
+const breadcrumbsTransformer = (context, node) => {
+  const componentName = node.openingElement.name;
+  const closingTag = node.closingElement.name;
+  const selectedFacilityNode = getPropNode(node, 'selectedFacility');
+
+  context.report({
+    node,
+    message: MESSAGE,
+    data: {
+      reactComponent: componentName.name,
+      webComponent: 'va-breadcrumbs',
+    },
+    suggest: [
+      {
+        desc: 'Migrate component',
+        fix: fixer => {
+          // Replace opening and close tags
+          // and remove `selectedFacilityNode` prop if present
+          return [
+            fixer.replaceText(componentName, 'va-breadcrumbs'),
+            fixer.replaceText(closingTag, 'va-breadcrumbs'),
+            selectedFacilityNode && fixer.remove(selectedFacilityNode),
+          ].filter(i => !!i);
+        },
+      },
+    ],
+  });
+};
+
+let isLibraryComponent = {};
+
+const updateIsLibraryImportObj = (context, componentName) => {
+  isLibraryComponent[context.getFilename()] = {
+    ...isLibraryComponent[context.getFilename()],
+    [componentName]: context
+      .getAncestors()[0]
+      .body.some(
+        node =>
+          node.type === 'ImportDeclaration' &&
+          node.source.value.includes(
+            `@department-of-veterans-affairs/component-library/${componentName}`,
+          ),
+      ),
+  };
+};
+
+const isLibraryImport = (context, componentName) => {
+  if (
+    !isLibraryComponent
+      .hasOwnProperty(context.getFilename())
+      ?.hasOwnProperty(componentName)
+  ) {
+    updateIsLibraryImportObj(context, componentName);
+  }
+  return isLibraryComponent[context.getFilename()][componentName];
+};
+
```

Comment:
This looks like a cool optimization, but can you add some comments to explain it a bit? Things that might be good to call out:

- What gets stored in `isLibraryComponent`, key and value
- How are we using `isLibraryComponent`